NuRadioMC/EvtGen/NuRadioProposal.py

```diff
@@ -303,14 +289,23 @@ def GetSecondariesArray(Eleptons, lepton_codes, random_seed=None):
     propagators = {}
     for lepton_code in np.unique(lepton_codes):
         if lepton_code not in propagators:
-            propagators[lepton_code] = create_propagator(low=low, particle_code=lepton_code)
+            propagators[lepton_code] = create_propagator(low=low, particle_code=lepton_code,
+                                                         config_file=config_file)
 
     secondaries_array = []
 
-    for Elepton, lepton_code in zip(Eleptons, lepton_codes):
+    if lepton_positions is None:
+        lepton_positions = [(0, 0, 0)] * len(Eleptons)
+    if lepton_directions is None:
+        lepton_directions = [(0, 0, -1)] * len(Eleptons)
 
-        propagators[lepton_code].particle.position = pp.Vector3D(0, 0, 0)
-        propagators[lepton_code].particle.direction = pp.Vector3D(0, 0, 1)
+    for Elepton, lepton_code, lepton_position, lepton_direction in zip(Eleptons,
```

Comment:
Same here. Does Elepton need to be capitalized for some reason?

Reply:
Because energy is traditionally represented by a capital E. Changed.